kimchi/src/tests/foreign_field_add.rs

```diff
@@ -162,7 +160,7 @@ fn test_ffadd(
         .iter()
         .map(|x| BigUint::from_bytes_be(x))
         .collect::<Vec<BigUint>>();
-    let witness = witness::create(&inputs, ops, foreign_modulus);
+    let witness = witness::create_ffadd_chain_witness(&inputs, ops, foreign_modulus, true);
```

Comment:
For reasons I described earlier, I would make this stuff with the test multi-range-checks part of the test code, rather than the actual gate's witness code.

Reply:
I see!

Reply:
in this case the code duplication is so huge that i don't feel it is any elegant..

Reply:
Did some refactor so there is no code duplication and became modular